src/dns/server.rs

```diff
@@ -656,14 +664,33 @@ struct SystemForwarder {
 }
 
 impl SystemForwarder {
-    fn new() -> Result<Self, Error> {
+    fn new(custom_name_servers: Vec<SocketAddr>) -> Result<Self, Error> {
```

Comment:
Do the changes here mean we hit our own internal DNS system for lookups? If so, interesting... we don't in Envoy. Part of me likes it (in memory lookup is obviously cheaper than a real one...). But worried we may end up with loops or something.

Will revist this comment as I read more

Reply:
This PR is still very WIP. This was for configuring a non system default dns name server so the nip.io style hosts resolve to known IPs in the direct test

Reply:
this code was removed